### PR TITLE
usm: http2: Split http2_frame_parser to http2_headers_parser and http2_eos_praser

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
The PR splits the program `socket__http2_frames_parser` into `socket__http2_headers_parser` and `socket__http2_eos_parser`.

The program `socket__http2_headers_parser` will be responsible on processing HEADER frames.
The program `socket__http2_eos_parser` will be responsible on processing End Of Stream frames, and RST frames, including enqueuing streams on the perf-event mechanism.

The PR allows us to theoretically process up to 468 frames for a single packet, instead of a theoretical limit of 203 today.
Today we can parse (both headers and EOS) up to 7 frames in a single tail call, and we have 29 free tail calls, thus we can process 203 (7*29).

After the change, in `socket__http2_headers_parser` we can parse up to 18 frames in a single tail call, and we can have up to 26 tail calls, thus 468 frames in total. While for `socket__http2_eos_parser` we can parse up to 200 frames in a single tail call, and 3 tail calls can give us a limit of 600 frames in total. We take a minimal value and get 468 as an upper bound.

At the moment, we can parse up to 120 (the current limit), so we adjust the number of tail calls allowed per program to respect that limit (7 for `socket__http2_headers_parser` and 1 for `socket__http2_eos_parser`).

### Motivation
The change serves us in 2 ways:
* Allows to bump the frames limit for parsing (a current goal of us)
* Reduces the complexity of `socket__http2_frames_parser` and allow us to add new code and features, without dealing with -EBIG from the verifier (as the program is too complicated for the verifer to track).

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h [hunk 1]
```diff
@@ -5,7 +5,19 @@
 
 #include "protocols/http2/defs.h"
 
-#define HTTP2_FRAMES_PER_TAIL_CALL 7
+// Represents the maximum number of frames we'll process in a single tail call in `handle_eos_frames` program.
+#define HTTP2_MAX_FRAMES_FOR_EOS_PARSER_PER_TAIL_CALL 200
+// Represents the maximum number of tail calls to process EOS frames.
+// Currently we have up to 120 frames in a packet, thus 1 tail call is enough.
+#define HTTP2_MAX_TAIL_CALLS_FOR_EOS_PARSER 1
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/decoding-defs.h:12]: Not sure I understood the use of HTTP2_MAX_TAIL_CALLS_FOR_EOS_PARSER in the current code change

File: pkg/network/ebpf/c/protocols/http2/decoding.h [hunk 2]
```diff
@@ -733,14 +697,14 @@ int socket__http2_filter(struct __sk_buff *skb) {
...
-        bpf_tail_call_compat(skb, &protocols_progs, PROG_HTTP2_FRAME_PARSER);
+        bpf_tail_call_compat(skb, &protocols_progs, PROG_HTTP2_HEADERS_PARSER);
     }
 
     return 0;
 }
 
-SEC("socket/http2_frames_parser")
-int socket__http2_frames_parser(struct __sk_buff *skb) {
+SEC("socket/http2_headers_parser")
```
Comment (amitslavin) [inline pkg/network/ebpf/c/protocols/http2/decoding.h:706]: maybe it is a good time to document our current programs and the callers in order to make some order ?

**Review by amitslavin — CHANGES_REQUESTED** [review]

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h (hunk 1 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/http2/decoding-defs.h:12]: The goal is to make the code ready to be called multiple times by just changing this value (similar behavior exists for the headers parser)

File: pkg/network/ebpf/c/protocols/http2/decoding.h (hunk 2 above)
Comment (guyarb) [reply pkg/network/ebpf/c/protocols/http2/decoding.h:706]: np

Comment (guyarb) [reply pkg/network/ebpf/c/protocols/http2/decoding.h:706]: I'll add documentation only for the new programs, and for the old ones I'll make that in a separate PR

File: pkg/network/ebpf/c/protocols/http2/decoding-defs.h (hunk 1 above)
Comment (amitslavin) [reply pkg/network/ebpf/c/protocols/http2/decoding-defs.h:12]: I see

**Review by amitslavin — APPROVED** [review]

**Pull request merged**